src/main/java/nus/climods/model/ModelManager.java

```diff
@@ -23,65 +23,103 @@
+
     private final ModuleList moduleList;
-    private final UniqueUserModuleList uniqueUserModuleList;
+    private final UniqueUserModuleList userModuleList;
 
-    private final FilteredList<UserModule> filteredUserModules;
-    private final UserPrefs userPrefs;
     private final FilteredList<Module> filteredModuleList;
+    private final SortedList<Module> filteredAndSortedModuleList;
+    private final Comparator<? super Module> defaultModuleListComparator;
+
+    private final FilteredList<UserModule> filteredUserModuleList;
+
+    private final UserPrefs userPrefs;
 
     /**
      * Initializes a ModelManager with the given moduleList and userPrefs.
      */
-    public ModelManager(ReadOnlyModuleList moduleList, UniqueUserModuleList uniqueUserModuleList,
+    public ModelManager(ReadOnlyModuleList moduleList, UniqueUserModuleList userModuleList,
                         ReadOnlyUserPrefs userPrefs) {
         requireAllNonNull(moduleList, userPrefs);
 
         logger.fine("Initializing with module list: " + moduleList + " and user prefs " + userPrefs);
 
         this.userPrefs = new UserPrefs(userPrefs);
         this.moduleList = new ModuleList(moduleList);
-        this.uniqueUserModuleList = uniqueUserModuleList;
+        this.userModuleList = userModuleList;
+
         this.filteredModuleList = new FilteredList<>(moduleList.getModules());
-        filteredUserModules = new FilteredList<UserModule>(uniqueUserModuleList.asUnmodifiableObservableList());
+        this.filteredAndSortedModuleList = new SortedList<>(filteredModuleList);
+        this.defaultModuleListComparator = filteredAndSortedModuleList.getComparator();
+
+        this.filteredUserModuleList = new FilteredList<>(userModuleList.asUnmodifiableObservableList());
     }
 
-    //=========== UserModule ==================================================================================
+    //=========== Module ==================================================================================
 
     @Override
-    public boolean hasUserModule(UserModule module) {
-        requireNonNull(module);
-        return uniqueUserModuleList.contains(module);
+    public ReadOnlyModuleList getModuleList() {
+        return moduleList;
     }
 
     @Override
-    public boolean filteredListhasUserModule(UserModule module) {
-        return this.getFilteredUserModuleList().contains(module);
+    public ObservableList<Module> getFilteredModuleList() {
+        return filteredAndSortedModuleList;
     }
 
     @Override
-    public void deleteUserModule(UserModule target) {
-        requireNonNull(target);
-        uniqueUserModuleList.remove(target);
+    public void setFilteredModuleList(Predicate<Module> predicate) {
+        requireNonNull(predicate);
+        this.filteredModuleList.setPredicate(predicate);
+        // Reset to default comparator
+        this.filteredAndSortedModuleList.setComparator(defaultModuleListComparator);
```

Comment:
We need to reset it so that a previous comparator will not persist and future filter result orderings